TextTemplatingFileGeneratorCore/TextTemplatingFileGeneratorCore.cs

```diff
@@ -99,28 +100,154 @@ private IEnumerable<string> ProcessReferences(string[] references, string inputF
+                        {
+                            projectPath = projectFile;
+                            projectDir = Path.GetDirectoryName(projectFile);
+                            if( projectDir != null && !projectDir.EndsWith(@"\") )
+                            {
+                                projectDir += @"\";
+                            }
+                            projectName = Path.GetFileNameWithoutExtension(project.FullName);
+                            projectExt = Path.GetExtension(project.FullName);
+                        }
+                        Configuration configuration = project.ConfigurationManager.ActiveConfiguration;
+                        if( configuration != null )
+                        {
+                            configurationName = configuration.ConfigurationName;
+                            platformName = configuration.PlatformName;
+                            string outputPath = (string)configuration.Properties.Item("OutputPath")?.Value;
+                            string outputFileName = (string)project.Properties.Item("OutputFileName")?.Value;
+                            if (outputPath != null && outputFileName != null)
+                            {
+                                targetFileName = outputFileName;
+                                targetName = Path.GetFileNameWithoutExtension(outputFileName);
+                                targetExt = Path.GetExtension(outputFileName);
+                                outDir = outputPath;
+                                if( !outDir.EndsWith(@"\") )
+                                {
+                                    outDir += @"\";
+                                }
+                                targetDir = projectDir + outDir;
+                                targetPath = targetDir + targetFileName;
+                            }
+                        }
+                    }
+                }
+            }
 
-            Solution solution = dte?.Solution;
-            Project project = solution?.FindProjectItem(inputFileName)?.ContainingProject;
-            Configuration configuration = project?.ConfigurationManager.ActiveConfiguration;
-
-            string solutionFile = solution?.FullName;
-            string solutionDir = solutionFile != null ? Path.GetDirectoryName(solutionFile) + "\\" : "";
-            string projectFile = project?.FullName;
-            string projectDir = projectFile != null ? Path.GetDirectoryName(projectFile) + "\\" : "";
+            var replacements = new Dictionary<string, string>
+            {
+                ["SolutionName"] = solutionName,             // "MySolution"
+                ["TargetName"] = targetName,                 // "MyProject"
+                ["ProjectName"] = projectName,               // "11.T4.MyProject"
+                ["ConfigurationName"] = configurationName,    // "Debug"
+                ["OutDir"] = outDir,                         // "bin\Debug\"
```

Comment:
Note: when I ran it against a .NET 5 project, it did say `bin\Debug\net5.0\` so it does include the actual output relative path as expected.